src/AutoRest.CSharp/Common/Input/Source/SourceInputHelper.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using AutoRest.CSharp.Generation.Types;
+using Microsoft.CodeAnalysis;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AutoRest.CSharp.Input.Source
+{
+    public static class SourceInputHelper
+    {
+        internal static IEnumerable<ITypeSymbol> GetSymbols(INamespaceSymbol namespaceSymbol)
+        {
+            foreach (var childNamespaceSymbol in namespaceSymbol.GetNamespaceMembers())
+            {
+                foreach (var symbol in GetSymbols(childNamespaceSymbol))
+                {
+                    yield return symbol;
+                }
+            }
+
+            foreach (INamedTypeSymbol symbol in namespaceSymbol.GetTypeMembers())
+            {
+                yield return symbol;
+            }
+        }
+
+        internal static bool IsEqualType(INamedTypeSymbol symbol, CSharpType type)
```

Comment:
I made some changes to this method [here](https://github.com/Azure/autorest.csharp/blob/6677daafde53094b1de56bba430c7a76f88e0b17/src/AutoRest.CSharp/Common/Utilities/NamedTypeSymbolExtensions.cs) (because I need it in another PR)
Could you take a look and apply those modifications here?
Also I think this method is better on the NamedTypeSymbolExtensions class as an extension class of `INamedTypeSymbol` therefore I move it there in my PR